src/adapter/src/coord/sequencer.rs

```diff
@@ -2184,9 +2184,13 @@ impl<S: Append + 'static> Coordinator<S> {
             let peek_ts = if matches!(
                 &peek_plan.plan,
                 peek::PeekPlan::FastPath(peek::FastPathPlan::Constant(_, _))
-            ) && timestamp_independent
-            {
-                None
+            ) {
+                if timestamp_independent {
+                    None
+                } else {
+                    //TODO(jkosh44) for constant timestamp dependent queries, we adopt the transactions timeline
+                    Some((peek_plan.timestamp, session.transaction().timeline()))
```

Comment:
What if this is the first query of the txn?

Reply:
If this is the first query of the transaction then `timeline()` will return `None` which should be equivalent to all constant queries timelines. I'll add an assert here for `assert!(timeline.is_none())`.

Reply:
Yeah, the current behavior is that if execute a constant timestamp dependent query in a transaction, then all subsequent queries must be constant. The motivating example here is that if you execute `SELECT mz_now()` as the first query in the transaction, then we must pick some timestamp which forces us into either a single timeline or no timeline. We currently choose no timeline, but maybe we should pick `EpochMillis`.


Executing a constant timestamp independent query as the first query in the transaction does not force us into a timeline. These are queries like `SELECT 1`. Queries like `SELECT * FROM pg_am` should be considered timestamp independent, which would not lock us into a timeline, but we have a bug where we consider it timestamp dependent.